File: alpro-lioneljevon-16-10-2025/keuangan-budi.py
print("=== Program Perhitungan Keuangan Budi Selama Liburan ===\n")

gaji_per_jam = float(input("Masukkan gaji per jam (Rp): "))
jam_minggu = float(input("Masukkan jumlah jam kerja per minggu: "))

def pendapatan_budi(gaji, jam):
    pendapatan_kotor = gaji*jam * 5
    pajak = 0.14 * pendapatan_kotor
    pendapatan_bersih = pendapatan_kotor - pajak
    return pendapatan_kotor, pajak, pendapatan_bersih

def pengeluaran_budi(pendapatan_bersih):
    pengeluaran_baju = 0.1 * pendapatan_bersih
    pengeluaran_alat_tulis = 0.01 * pendapatan_bersih
    sisa_belanja = pendapatan_bersih - (pengeluaran_baju + pengeluaran_alat_tulis)
    return pengeluaran_baju, pengeluaran_alat_tulis, sisa_belanja

def sedekah_budi(sisa_belanja):
    sedekah = 0.25 * sisa_belanja
    sedekah_yatim = 0.3 * sedekah
    sedekah_dhuafa = sedekah - sedekah_yatim
    return sedekah, sedekah_yatim, sedekah_dhuafa

pendapatan_kotor, pajak, pendapatan_bersih = pendapatan_budi(gaji_per_jam, jam_minggu)
pengeluaran_baju, pengeluaran_alat_tulis, sisa_belanja = pengeluaran_budi(pendapatan_bersih)
sedekah, sedekah_yatim, sedekah_dhuafa = sedekah_budi(sisa_belanja)

print("\nRincian Keuangan Budi ")
print(f"Pendapatan Kotor: Rp {pendapatan_kotor:,.0f}".replace(",", ".") )
print(f"Pajak (14%): Rp {pajak:,.0f}".replace(",", ".") )
print(f"Pendapatan Bersih: Rp {pendapatan_bersih:,.0f}".replace(",", ".") )
print(f"Pengeluaran untuk Baju: Rp {pengeluaran_baju:,.0f}".replace(",", ".") )
print(f"Pengeluaran untuk Alat Tulis: Rp {pengeluaran_alat_tulis:,.0f}".replace(",", ".") )
print(f"Sisa Belanja: Rp {sisa_belanja:,.0f}".replace(",", ".") )
print(f"Total Sedekah (25% dari Sisa Belanja): Rp {sedekah:,.0f}".replace(",", ".") )
print(f"  - Sedekah untuk Anak Yatim (30% dari Total Sedekah): Rp {sedekah_yatim:,.0f}".replace(",", ".") )
print(f"  - Sedekah untuk Dhuafa: Rp {sedekah_dhuafa:,.0f}".replace(",", ".") )

File: alpro-lioneljevon-16-10-2025/Versi C++/keuangan-budi.cpp
#include <iostream>
#include <iomanip>
using namespace std;

void pendapatan_budi(double gaji, double jam, double &pendapatan_kotor, double &pajak, double &pendapatan_bersih) {
    pendapatan_kotor = gaji * jam * 5;
    pajak = 0.14 * pendapatan_kotor;
    pendapatan_bersih = pendapatan_kotor - pajak;
}

void pengeluaran_budi(double pendapatan_bersih, double &pengeluaran_baju, double &pengeluaran_alat_tulis, double &sisa_belanja) {
    pengeluaran_baju = 0.1 * pendapatan_bersih;
    pengeluaran_alat_tulis = 0.01 * pendapatan_bersih;
    sisa_belanja = pendapatan_bersih - (pengeluaran_baju + pengeluaran_alat_tulis);
}

void sedekah_budi(double sisa_belanja, double &sedekah, double &sedekah_yatim, double &sedekah_dhuafa) {
    sedekah = 0.25 * sisa_belanja;
    sedekah_yatim = 0.3 * sedekah;
    sedekah_dhuafa = sedekah - sedekah_yatim;
}

int main() {
    cout << "=== Program Perhitungan Keuangan Budi Selama Liburan ===\n" << endl;

    double gaji_per_jam, jam_minggu;
    cout << "Masukkan gaji per jam (Rp): ";
    cin >> gaji_per_jam;
    cout << "Masukkan jumlah jam kerja per minggu: ";
    cin >> jam_minggu;

    double pendapatan_kotor, pajak, pendapatan_bersih;
    double pengeluaran_baju, pengeluaran_alat_tulis, sisa_belanja;
    double sedekah, sedekah_yatim, sedekah_dhuafa;

    pendapatan_budi(gaji_per_jam, jam_minggu, pendapatan_kotor, pajak, pendapatan_bersih);
    pengeluaran_budi(pendapatan_bersih, pengeluaran_baju, pengeluaran_alat_tulis, sisa_belanja);
    sedekah_budi(sisa_belanja, sedekah, sedekah_yatim, sedekah_dhuafa);

    cout << "\nRincian Keuangan Budi" << endl;
    cout << fixed << setprecision(0);
    cout.imbue(locale(""));

    cout << "Pendapatan Kotor: Rp " << pendapatan_kotor << endl;
    cout << "Pajak (14%): Rp " << pajak << endl;
    cout << "Pendapatan Bersih: Rp " << pendapatan_bersih << endl;
    cout << "Pengeluaran untuk Baju: Rp " << pengeluaran_baju << endl;
    cout << "Pengeluaran untuk Alat Tulis: Rp " << pengeluaran_alat_tulis << endl;
    cout << "Sisa Belanja: Rp " << sisa_belanja << endl;
    cout << "Total Sedekah (25% dari Sisa Belanja): Rp " << sedekah << endl;
    cout << "  - Sedekah untuk Anak Yatim (30% dari Total Sedekah): Rp " << sedekah_yatim << endl;
    cout << "  - Sedekah untuk Dhuafa: Rp " << sedekah_dhuafa << endl;

    return 0;
}
